ClimbingPlaylistApi/Services/PlaylistService.cs

```diff
@@ -9,12 +10,135 @@ namespace ClimbingPlaylistApi.Services
 {
     public class PlaylistService : IPlaylistService
     {
-        public PlaylistService()
+        // TODO: update to stateless?
+        public PlaylistService(IDbService dbService, IRouteModelHandler routeHandler)
         {
+            _dbService = dbService;
+            _routeHandler = routeHandler;
+            OnInitialize();
+        }
+
+        private void OnInitialize()
+        {
+            Playlists = _dbService.GetAllPlaylists();
+        }
+
+        private IRouteModelHandler _routeHandler { get; set; }
+        private IDbService _dbService { get; set; }
 
+        public List<PlaylistModel> Playlists { get; set; } = new List<PlaylistModel>();
+
+        public void AddPlaylist(PlaylistModel playlist)
+        {
+            if (Playlists.Any(p => p.Name == playlist.Name))
+            {
+                var message = $"Playlist with name \"{playlist.Name}\" already exists.";
+                throw new ArgumentException(message);
+            }
+            Playlists.Add(playlist);
```

Comment:
I think a more appropriate way to do this would be to totally reload your 'local' playlist array from the DB